SafeguardSessions/test/Unit/version/TestCheckVersion.query.pq

```diff
@@ -0,0 +1,70 @@
+section TestCheckVersion;
+
+Version = Extension.ImportModule("Version.pqm");
+
+Version.CheckVersion = Version[CheckVersion];
+
+SupportedVersions.Versions = Extension.ImportFunction("Versions", "SupportedVersions.pqm");
+
+LatestSupportedVersion = UnitTestFramework.ImportFunction("Version", "LatestSupportedVersion.pqm");
+
+AssertVersionCheck = (description as text, version as text, supportedVersions as list, expectedValue as logical) =>
+    Fact(description, expectedValue, Version.CheckVersion(version, supportedVersions));
+
+TestCheckVersion = () =>
+    let
+        dummySupportedVersions = {"custom", "1.1", "2.0", "2.1"},
+        cases = {
+            {
+                "Custom SPS version for testing is supported by the custom connector",
+                "custom_version_for_testing",
+                dummySupportedVersions,
+                true
+            },
+            {
+                "Latest supported SPS version is supported by the custom connector",
+                "2.1",
+                dummySupportedVersions,
+                true
+            },
+            {
+                "Future supported SPS version without breaking change will be supported by the custom connector",
+                "2.1.1",
+                dummySupportedVersions,
+                true
+            },
+            {"Supported LTS SPS version is supported by the custom connector", "2.0", dummySupportedVersions, true},
+            {
+                "Future supported LTS version without breaking change will be supported by the custom connector",
+                "2.0.1",
+                dummySupportedVersions,
+                true
+            },
+            {"Supported older SPS version is supported by the custom connector", "1.1", dummySupportedVersions, true},
+            {
+                "Future SPS version with breaking change is not supported by the custom connector",
+                "2.2",
+                dummySupportedVersions,
+                false
+            },
+            {
+                "Unsupported LTS SPS version is not supported by the custom connector",
+                "1.0",
+                dummySupportedVersions,
+                false
+            }
+        },
+        facts = ProvideDataForTest(cases, AssertVersionCheck)
+    in
+        facts;
+
+TestLatestSPSVersionIsSupported = () =>

```

Comment:
Okay. It missed my attention. I'm afraid that we will forget to update the test asset version and this test will become deprecated.

It this is an expectional, we are indeed better to keep the LatestVersion in SupportedVersions.pqm where we won't forget to update it.

Reply:
I'll move it back. ;)

Reply:
Done.